pdc_client/__init__.py

```diff
@@ -237,3 +249,58 @@ def set_comment(self, comment):
         :paramtype comment: string
         """
         self.session.headers["PDC-Change-Comment"] = comment
+
+
+class _BeanBagWrapper(object):
+    """
+       Wrapper of BeanBag's attributes and items.
+
+       This class wraps attributes and items of Beanbag to let page_size of
+       PDCClient's constructor work.
+    """
+    def __init__(self, client, page_size, omit_paging=False):
+        self.client = client
+        self.page_size = page_size
+        self.omit_paging = omit_paging
+
+    def __call__(self, *args, **kwargs):
+        if 'page_size' not in kwargs:
+            kwargs['page_size'] = self.page_size
+
+        result = self.client(*args, **kwargs)
+        if self.omit_paging and self.is_multi_page_response(result) and ("page" not in kwargs.keys()):
+            count = result["count"]
+            multi_result = result['results']
+            while result['next']:
+                next_page = self.url2path_ext(result['next'])
+                kwargs["page"] = next_page["page"][0]
+                result = self.client(*args, **kwargs)
+                multi_result.extend(result['results'])
```

Comment:
This seems to work the same way as `page_size=-1` but is slower since this sends multiple requests.

Reply:
I just think people will write a for loop to get all the data. But currently we just return the whole data.
But as you said return as a iterator saves more space of pdc_client, that makes sense,  have changed the codes